worklist/views.py

```diff
@@ -1,36 +1,87 @@
+import functools
 import json
-from django.http import JsonResponse
+from django.http import JsonResponse, HttpResponseRedirect
 from django.shortcuts import render
 import logging
 from worklist.constants import SEARCH_BY_NAME_FOR_WORKLISTS, \
-    SEARCH_BY_USERNAME_FOR_WORKLISTS
+    SEARCH_BY_USERNAME_FOR_WORKLISTS, ARTICLE_STATUS_TO_NUMBER_MAPPING, \
+    ARTICLE_NUMBER_TO_STATUS_MAPPING
 from worklist.views_helper_functions.store_articles import \
     store_added_articles, store_psid_articles
 from worklist.models import WorkList, Task
+from django.contrib.auth.decorators import login_required
+from django.contrib.auth import logout
+
 
 logger = logging.getLogger('django')
 
 
+# View related to OAuth using MediaWiki
+def mediawiki_login(request):
+    return HttpResponseRedirect(
+        'https://tools.wmflabs.org/worklist-tool/oauth/login/mediawiki/')
+
+
+# View for logging out from the app
+@login_required
+def app_logout(request):
+    logout(request)
+
+    # if user has logged out of the app, take him to home page
+    return render(request, 'homepage.html',
+                  {'message': '',
+                   'is_logged_in': False,
+                   'error': False,
+                   'logged_in_user': '',
+                   })
+
+
+def with_logged_in_user(view):
+
+    @functools.wraps
+    def inner_method(request):
+        user_or_none = None
+        is_logged_in = False
+
+        if request.user.is_authenticated:
+            user_or_none = request.user.username
+            is_logged_in = True
+
+        return view(request, user_or_none, is_logged_in)
```

Comment:
Optional, but I'd just skip the extra `is_logged_in` boolean -- the wrapped function knows that the username can be None, and in that case we're logged out.

Reply:
Done.